forem/forem
# Remove redis-rails gem

## What type of PR is this? (check all applicable)


## Description

While working on the [Rails 7 upgrade](https://github.com/forem/forem/pull/15908) I had issues with several Redis-related packages holding back the update. So I dug a bit deeper into [redis-rails](https://github.com/redis-store/redis-rails) and noticed the following:

> Rails 5.2.0 includes a Redis cache store out of the box, so you don't really need this gem anymore if you just need to store the fragment cache in Redis. 

I did a quick check and our fragment cache is indeed set up to use the built-in `:redis_cache_store` and not the `:redis_store` provided by this gem.

However, we do also use Redis as our session store. Again, the README had an important pointer here:

> If you need session storage, consider directly using redis-actionpack instead.

We're already using this gem, in fact, we have it (and `redis-rack`) pinned to specific versions (I updated the comment as part of this PR):

```ruby
# NOTE: [@rhymes, @citizen428]: sign in doesn't work with redis-rack 2.1.0+ or with redis-actionpack 5.2.0+
# We need to investigate why exactly, in the meantime we are hardcoding these two dependencies
# to keep them at the latest working version
gem "redis-actionpack", "5.1.0" # Redis session store for ActionPack. Used for storing the Rails session in Redis.
gem "redis-rack", "2.0.6" # Redis Store for Rack applications
```

For now, I just want to remove this apparently unneeded gem and its direct dependency `redis-activesupport`. After that, I can look into upgrading `redis-actionpack` and `redis-rack`.

## Related Tickets & Documents

https://github.com/forem/forem/issues/15825

## QA Instructions, Screenshots, Recordings

* All existing specs should be green
* Start a Rails server locally and ensure that you can log in

### UI accessibility concerns?

No UI changes

## Added/updated tests?

- [X] No, and this is why: this PR just removes an unnecessary dependency, no code/test changes should be necessary

## [Forem core team only] How will this change be communicated?

- [X] This change does not need to be communicated, and this is why not: this PR just removes an unnecessary dependency
